include/slate/BaseMatrix.hh

```diff
@@ -1787,14 +1791,34 @@ void BaseMatrix<scalar_t>::tileSend(
 ///     Pointer to an MPI_Request struct
 ///
 template <typename scalar_t>
-template <Target target>
 void BaseMatrix<scalar_t>::tileIsend(
     int64_t i, int64_t j, int dst_rank, int tag, MPI_Request* request)
 {
     if (dst_rank != mpiRank()) {
-        //todo: need to qcquire read access lock to TileNode(i, j)
-        tileGetForReading(i, j, LayoutConvert::None);
-        at(i, j).isend(dst_rank, mpiComm(), tag, request);
+        //todo: need to acquire read access lock to TileNode(i, j)
+
+        int send_dev = HostNum;
+        if (gpu_aware_mpi()) {
+            // If we have GPU aware MPI, we can use any existing valid tile
+            auto& tile_node = storage_->at(globalIndex(i, j));
+            // acquire read access to the (i, j) TileNode
+            LockGuard guard(tile_node.getLock());
+
+            // find a valid source (Modified/Shared) tile
+            for (int d = HostNum; d < num_devices(); ++d) {
```

Comment:
Should this prefer a GPU device, if it exists there? I guess on Frontier it should, but on Summit-like systems perhaps not.